tests/integration/suite/daprd/serviceinvocation/http/httpendpoints.go

```diff
@@ -0,0 +1,161 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	procdaprd "github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(httpendpoints))
+}
+
+type httpendpoints struct {
+	daprd1  *procdaprd.Daprd
+	appPort int
+}
+
+func (h *httpendpoints) Setup(t *testing.T) []framework.Option {
+	newHTTPServer := func() *prochttp.HTTP {
+		handler := http.NewServeMux()
+
+		handler.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("ok"))
+		})
+
+		return prochttp.New(t, prochttp.WithHandler(handler))
+	}
+
+	newHTTPServerTLS := func() *prochttp.HTTP {
+		handler := http.NewServeMux()
+
+		handler.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("ok"))
+		})
+
+		privKey := "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCJFS3cMM+6e3KZ\nPriLhkvzU6TE0TWGAZz08Z8I4GOTms/wsGBFr3mZoFNsrkZg0kOo+0PoC1WlXcPZ\n21zhDQJn/y2sUGfRypi41yGjXd/rLRjyIfsz1GsCj1dnVQDCGhJP0ZwH7DdnhVE+\nww0FuUcmmlr0EaAfsupkQhlE9QKyfrZE4WPF/HumC9vLbvpaNw/95cumctUilVMx\nK04YpyLtSXShmRvXjVUej/zS6HrDOFbu05lAO+E6BH+p747+n08EV75FOj/5iFYM\npF0+2Nt0QXsqJW4od9AlHaCsL2wf8rubpLrzbyKmbDqQpxKpB9749ReEfIVBKymK\nCR7JwstTAgMBAAECggEACdHywUGH6AqG456AIZKjJzvDOSGUQtTOayenURSjXYQW\nHbPCOcu/xiHGSCoAkFgPu8UVkMPFxL0VANV7mlhQZ1B686CmR1D1pCwufzcENE3n\nPxVFxb6TqeZzMvZTwb6UuD4XqOOi9m9bqhzWoSXW3deUk25zI5eGvgbpgNdHya86\nRLZgEDGSnnAMZi7FANXR+gtvHQmXYP9CM/1rivYQI/MXd4jbZEVWeNZEc1u1YqVC\nM7ddN063KmE/LQewF2dHhw5dyhh1udgpoXuOFhmkbwBPLn6eVJ/GdKRG0+LDfOmW\nFx6QN7HV0gYISdVIUQBOsEPSh2LEurKDJyqzFLUoiQKBgQC+B08abD4D+3JQTmPV\nZuHusL3wizZxx4YQ1SNULMOv+i7oP22ERXsCCrnRGn+JPM/jc2HV0I3Z67BS9a2M\nUjRVZ54rN5DTWP+XAN1rCcDkxrp0CZ9pL4UZwcAZyanIIYCc0RcTHM//2WblLtHg\n60OJ5lMTlqc2LhQquwIM/5iAKwKBgQC4rFZYXoj6AYkdEyQSWcqOrtH72h5GZ9Rw\nmETRk8GeMI2gD6t92DWZmZcC+ycUGDxBX1A7fQ+imw31KUz0ciAX+aDWEgUBOgS5\n50KvRAo9JcHxyj5qXPlXpCHItaGjd2A1KexvKRxN4GngEouQcOTSF69pyP4P837v\nFfVclHgleQKBgHa8rmq+M8ndNcKEGNFhJc81VJmXAv/5QgYGh7qy9dthoimwaEf7\n+i5+cTj9K6+e9e8TS5CEbf46zMQpirVhMB4lTqmGYNIOjDXYboHRaFwK6LpUwOzC\nqpI7hEMYxSOt+2UBKT/iAv3E5AxUQwQdPOhPqJ1Wx0iNZkCE9XUVyD5dAoGADY6B\nLC2MaqXwNdUw4bP7vauxuIZTkKGQo89ZxfTN0toHm4dq9GsJzEPNJSfgv4Xj7nyb\nvDI7EpnFVYj5oDw8hOYLW4upGGT08dy7NXiOM9zwtto86Lv4hemDnWNQAVsDEgQI\n2kQvUrw1qYBTBIB1G8MnWmGp3OvoFo8LGwe/JikCgYA9RgIuXNkcYMPtRuOzX/ne\nCxP3enIwGgTfMZl7i4wprbv11FwoihhhkPjnY/tQikvmw6zTkqPrnALdDia15kmt\n0YpGrkkFOQ1RzPP4brOr17mcA2RQqfzWxq/trXiJVVIch7c4EBliUCH+qr7WMGUK\nb0K2SICFbq1Z4B4X9/MqGw==\n-----END PRIVATE KEY-----\n"
```

Comment:
Nit for the future... Conside rmoving these outside of the function, maybe in a `keys.go` file, so it makes the code easier to read without large blocks of base64-encoded stuff.

PS: P-256 or Ed25519 keys and certificates are a lot shorter :)

Reply:
Ok